apps/addons/models.py

```diff
@@ -358,8 +358,19 @@ class Addon(amo.models.OnChangeMixin, amo.models.ModelBase):
     # Whether the add-on is listed on AMO or not.
     is_listed = models.BooleanField(default=True, db_index=True)
 
-    objects = AddonManager()
+    # The order of those managers is very important:
+    # The first one discovered, if it has "use_for_related_fields = True"
+    # (which it has if it's inheriting from caching.base.CachingManager), will
+    # be used for relations like `version.addon`. We thus want one that is NOT
+    # filtered in any case, we don't want a 500 if the addon is not found
+    # (because it's deleted for example).
+    # The CLASS of the first one discovered will also be used for "many to many
+    # relations" like `collection.addons`. In that case, we do want the
+    # filtered version by default, to make sure we're not displaying stuff by
+    # mistake. You thus want the CLASS of the first one to be filtered by
+    # default (that's why we have the "include_deleted=False" named param).
     with_deleted = AddonManager(include_deleted=True)
+    objects = AddonManager(include_deleted=False)
```

Comment:
Is `include_deleted=False` really needed ? It's the default value for this keyword argument.